app/src/main/java/com/rootstrap/android/ui/activity/main/targetpoint/CreateTargetView.kt

```diff
@@ -1,34 +1,102 @@
 package com.rootstrap.android.ui.activity.main.targetpoint
 
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
 import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.LinearLayoutManager
 import com.google.android.material.bottomsheet.BottomSheetBehavior
 import com.rootstrap.android.databinding.ActivityTargetPointsBinding
 import com.rootstrap.android.network.models.Target
+import com.rootstrap.android.network.models.Topic
+import com.rootstrap.android.ui.adapter.TopicAdapter
 import com.rootstrap.android.util.extensions.value
 import kotlinx.android.synthetic.main.fragment_create_target.view.*
+import kotlinx.android.synthetic.main.layout_topics.view.*
 
 class CreateTargetView(
     private val binding: ActivityTargetPointsBinding,
-    private val targetPointsViewModel: TargetPointsViewModel
+    private val targetPointsViewModel: TargetPointsViewModel,
+    private val lifecycleOwner: LifecycleOwner
 ) {
 
     private val bottomSheetBehavior: BottomSheetBehavior<ConstraintLayout> =
         BottomSheetBehavior.from<ConstraintLayout>(binding.root.create_target_bottom_sheet)
 
+    private val topicsBottomSheetBehavior: BottomSheetBehavior<ConstraintLayout> =
+        BottomSheetBehavior.from<ConstraintLayout>(binding.root.select_topic_layout)
+
+    private lateinit var topicAdapter: TopicAdapter
+
+    private var selectedTopic: Topic? = null
+
     init {
         bottomSheetBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
         bottomSheetBehavior.peekHeight = TargetPointsActivity.PICK_HEIGHT_HIDDEN
+
+        initTopicsBottomSheet()
+
+        binding.root.topic_edit_text.onFocusChangeListener = View.OnFocusChangeListener { v, hasFocus ->
+            if (hasFocus)
+                expandTopic()
+        }
+
+        binding.root.topic_edit_text.setOnClickListener {
+            expandTopic()
+        }
+
         binding.root.save_target_btn.setOnClickListener {
             createTarget()
         }
-        targetPointsViewModel.getTopics()
+
+        getTopics()
+
+        initTopicsFilter()
+    }
+
+    private fun getTopics() {
+        targetPointsViewModel.getTopics().observe(lifecycleOwner, Observer {
+            initTopicList(it)
+        })
+    }
+
+    private fun initTopicsBottomSheet() {
+        topicsBottomSheetBehavior.state = BottomSheetBehavior.STATE_HIDDEN
+        topicsBottomSheetBehavior.peekHeight = TargetPointsActivity.PICK_HEIGHT_HIDDEN
+
+        topicsBottomSheetBehavior.addBottomSheetCallback(object : BottomSheetBehavior.BottomSheetCallback() {
```

Comment:
use the socpe function apply to work over the same scope: `topicsBottomSheetBehavior.apply {...}`